### PR TITLE
gstreamer: set size to appsrc if instantiated

### PR DESCRIPTION
**Pull request by ceyusa**

### CODE REVIEW

**Comment by Manishearth** [comment]
r=me, waiting for travis

**Comment by ferjm** [comment]
It seems that we can remove [PlayerInner.input_size](https://github.com/servo/media/blob/0eb8ae51de1215e99498365bc3778e43698b948e/backends/gstreamer/src/player.rs#L85)

File: backends/gstreamer/src/player.rs [hunk 1]
```diff
@@ -195,6 +195,13 @@ impl Player for GStreamerPlayer {
 
     fn set_input_size(&self, size: u64) {
         self.inner.lock().unwrap().set_input_size(size);
+        if let Some(ref mut appsrc) = self.inner.lock().unwrap().appsrc {
+            if size > 0 {
+                appsrc.set_size(size as i64);
```
[the remark this answers is not in the document]
Comment (ceyusa) [reply backends/gstreamer/src/player.rs]: > It seems that we can remove PlayerInner.input_size

If the user sets the input_size() before calling setup() the data will be lost.

> Should we also set max-bytes here? (i.e. 10% of size)

I experimented with it yesterday, but it didn't force a quicker metadata extraction. In my opinion, an EOF in the network doesn't mean a EOS in the video processing. We would wait for EOS or an error from the player.

[the remark this answers is not in the document]
Comment (ferjm) [reply backends/gstreamer/src/player.rs]: > If the user sets the input_size() before calling setup() the data will be lost.

Fair enough. Let's add a comment about that to make sure that we don't remove it in the future.

[the remark this answers is not in the document]
Comment (ferjm) [reply backends/gstreamer/src/player.rs]: > I experimented with it yesterday, but it didn't force a quicker metadata extraction. In my opinion, an EOF in the network doesn't mean a EOS in the video processing. We would wait for EOS or an error from the player.

For reference, this was fixed by https://github.com/servo/servo/pull/21543/commits/1c73c720af71aa304eefb286a2816f72bf334277

**Comment by ferjm** [comment]
It seems that we broke the audio decoder at some point #136

**Pull request merged**